crates/forge/src/runner.rs

```diff
@@ -174,14 +174,54 @@ impl<'a> ContractRunner<'a> {
             traces.extend(setup_traces.map(|traces| (TraceKind::Setup, traces)));
             logs.extend(setup_logs);
 
-            TestSetup { address, logs, traces, labeled_addresses, reason, coverage }
+            TestSetup {
+                address,
+                logs,
+                traces,
+                labeled_addresses,
+                reason,
+                coverage,
+                fuzz_fixtures: self.fuzz_fixtures(address),
+            }
         } else {
-            TestSetup::success(address, logs, traces, Default::default(), None)
+            TestSetup::success(
+                address,
+                logs,
+                traces,
+                Default::default(),
+                None,
+                self.fuzz_fixtures(address),
+            )
         };
 
         Ok(setup)
     }
 
+    /// Collect fixtures from test contract. Fixtures are functions prefixed with `fixture_` key
+    /// and followed by the name of the parameter.
+    ///
+    /// For example:
+    /// `fixture_test() returns (address[] memory)` function
+    /// define an array of addresses to be used for fuzzed `test` named parameter in scope of the
+    /// current test.
+    fn fuzz_fixtures(&mut self, address: Address) -> FuzzFixtures {
+        // collect test fixtures param:array of values
+        let mut fixtures = HashMap::new();
+        self.contract.functions().for_each(|func| {
+            if func.name.is_fixture() {
+                if let Ok(CallResult { raw: _, decoded_result }) =
+                    self.executor.call(CALLER, address, func, &[], U256::ZERO, None)
+                {
+                    fixtures.insert(
+                        func.name.strip_prefix("fixture_").unwrap().to_string(),
+                        decoded_result,
```

Comment:
we should check for validity of the decoded result here instead of later when running the tests since all necessary info to do this is available here

Reply:
that would be nice to have but needs special handling of handler based testing (where fuzzed functions are not in same contract that defines `setUp/invariant_/fixture_` functions). It can be done but it adds code complexity to look up such info in all contracts created when `setUp`, wdyt?